src/icpbin_backend/src/lib.rs

```diff
@@ -235,12 +229,12 @@ fn create_new_paste(value: PasteDataCreator) -> Result<PasteData, IcpPasteError>
     let short = short_url.clone().unwrap_or("".to_string());
 
     // check short_url length and
-    if short_url.is_some() && (short.len() < MIN_SHORT_SIZE || short.len() > MAX_SHORT_SIZE) {
+    if short.len() < MIN_SHORT_SIZE || short.len() > MAX_SHORT_SIZE {
         return Err(IcpPasteError::ShortUrlShouldBeBetween4And10);
     }
 
     // short url should be unique
-    if short_url.is_some() && _is_short_url_exist(&short) {
+    if _is_short_url_exist(&short) {
```

Comment:
this way has performance issue. if url doesn't exist it check for empty string in short.
why we add extra search in blockchain data while we know it is empty